services/f3/driver/pull_request.go

```diff
@@ -0,0 +1,316 @@
+	FetchFunc func(repository string) string
+}
+
+func PullRequestConverter(f *issues_model.PullRequest) *PullRequest {
+	return &PullRequest{
+		PullRequest: *f,
+	}
+}
+
+func (o PullRequest) GetID() int64 {
+	return o.Index
+}
+
+func (o *PullRequest) SetID(id int64) {
+	o.Index = id
+}
+
+func (o *PullRequest) IsNil() bool {
+	return o.Index == 0
+}
+
+func (o *PullRequest) Equals(other *PullRequest) bool {
+	return o.Issue.Title == other.Issue.Title
+}
+
+func (o PullRequest) IsForkPullRequest() bool {
+	return o.HeadRepoID != o.BaseRepoID
+}
+
+func (o *PullRequest) ToFormat() *format.PullRequest {
+	var milestone string
+	if o.Issue.Milestone != nil {
+		milestone = o.Issue.Milestone.Name
+	}
+
+	labels := make([]string, 0, len(o.Issue.Labels))
+	for _, label := range o.Issue.Labels {
+		labels = append(labels, label.Name)
+	}
+
+	var mergedTime *time.Time
+	if o.HasMerged {
+		mergedTime = o.MergedUnix.AsTimePtr()
+	}
+
+	getSHA := func(repo *repo_model.Repository, branch string) string {
+		r, err := git.OpenRepository(context.Background(), repo.RepoPath())
+		if err != nil {
+			panic(err)
+		}
+		defer r.Close()
+
+		b, err := r.GetBranch(branch)
+		if err != nil {
+			panic(err)
+		}
+
+		c, err := b.GetCommit()
+		if err != nil {
+			panic(err)
+		}
+		return c.ID.String()
+	}
+
+	head := format.PullRequestBranch{
+		CloneURL:  o.HeadRepo.CloneLink().HTTPS,
+		Ref:       o.HeadBranch,
+		SHA:       getSHA(o.HeadRepo, o.HeadBranch),
+		RepoName:  o.HeadRepo.Name,
+		OwnerName: o.HeadRepo.OwnerName,
+	}
+
+	base := format.PullRequestBranch{
+		CloneURL:  o.BaseRepo.CloneLink().HTTPS,
+		Ref:       o.BaseBranch,
+		SHA:       getSHA(o.BaseRepo, o.BaseBranch),
+		RepoName:  o.BaseRepo.Name,
+		OwnerName: o.BaseRepo.OwnerName,
+	}
+
+	return &format.PullRequest{
+		Common:      format.Common{Index: o.Index},
+		PosterID:    o.Issue.Poster.ID,
+		PosterName:  o.Issue.Poster.Name,
+		PosterEmail: o.Issue.Poster.Email,
+		Title:       o.Issue.Title,
+		Content:     o.Issue.Content,
+		Milestone:   milestone,
+		State:       string(o.Issue.State()),
+		IsLocked:    o.Issue.IsLocked,
+		Created:     o.Issue.CreatedUnix.AsTime(),
+		Updated:     o.Issue.UpdatedUnix.AsTime(),
+		Closed:      o.Issue.ClosedUnix.AsTimePtr(),
+		Labels:      labels,
+		//		Assignees:      assignees,
+		PatchURL:       o.Issue.PatchURL(),
+		Merged:         o.HasMerged,
+		MergedTime:     mergedTime,
+		MergeCommitSHA: o.MergedCommitID,
+		Head:           head,
+		Base:           base,
+	}
+}
+
+func (o *PullRequest) FromFormat(pullRequest *format.PullRequest) {
+	labels := make([]*issues_model.Label, 0, len(pullRequest.Labels))
+	for _, label := range pullRequest.Labels {
+		labels = append(labels, &issues_model.Label{Name: label})
+	}
+
+	// assignees := make([]*user_model.User, 0, len(pullRequest.Assignees))
+	// for _, a := range pullRequest.Assignees {
+	// 	assignees = append(assignees, &user_model.User{
+	// 		Name: a,
+	// 	})
+	// }
```

Comment:
No commented-out code please, and if you really must, add a comment on the reason.

Reply:
Good catch, it was an oversight.